### PR TITLE
[frontend] Massive operations / checkboxes in Arsenal (Channels / Tools / Vulnerabilities) (#4234)

### LINKED ISSUE #4234: Massive operations / checkboxes in Arsenal (Channels / Tools / Vulnerabilities)
## Use case

Massive operations / checkboxes in Arsenal (Channels / Tools / Vulnerabilities)

### PR DESCRIPTION
**Pull request by Gwendoline-FAVRE-FELIX**
## Proposed changes

Possibility of massive operations on : 
  - Channels
  - Tools
  - Vulnerabilities
  
## Related issues 

- https://github.com/OpenCTI-Platform/opencti/issues/4234

### CODE REVIEW

**Comment by Archidoit** [comment]
- columns names and columns contents are not aligned : 
image

- empty values should be displayed as '-' (see empty 'types' in the screenshot)

**Comment by Archidoit** [comment]
'add' buttons overlaps the toolbar : 
image

**Comment by Gwendoline-FAVRE-FELIX** [comment]
The changes have been made

File: opencti-platform/opencti-front/src/private/components/arsenal/Channels.tsx [hunk 1]
```diff
@@ -81,61 +95,85 @@ const Channels = () => {
...
-    <>
+    <ExportContextProvider>
       <Breadcrumbs variant="list" elements={[{ label: t_i18n('Arsenal') }, { label: t_i18n('Channels'), current: true }]} />
       {renderLines()}
-      <Security needs={[KNOWLEDGE_KNUPDATE]}>
-        <ChannelCreation paginationOptions={queryPaginationOptions} />
-      </Security>
-    </>
+      {!FAB_REPLACED
+            && <Security needs={[KNOWLEDGE_KNUPDATE]}>
```
Comment (SarahBocognano) [inline opencti-platform/opencti-front/src/private/components/arsenal/Channels.tsx]: Indent a little ? :)

File: opencti-platform/opencti-front/src/private/components/arsenal/Tools.tsx [hunk 2]
```diff
@@ -71,60 +89,84 @@ const Tools = () => {
...
-    <>
+    <ExportContextProvider>
       <Breadcrumbs variant="list" elements={[{ label: t_i18n('Arsenal') }, { label: t_i18n('Tools'), current: true }]} />
       {renderLines()}
-      <Security needs={[KNOWLEDGE_KNUPDATE]}>
-        <ToolCreation paginationOptions={queryPaginationOptions} />
-      </Security>
-    </>
+      {!FAB_REPLACED
+            && <Security needs={[KNOWLEDGE_KNUPDATE]}>
```
Comment (SarahBocognano) [inline opencti-platform/opencti-front/src/private/components/arsenal/Tools.tsx]: Same here (indent)

File: opencti-platform/opencti-front/src/private/components/arsenal/tools/ToolLine.tsx [hunk 3]
```diff
@@ -88,26 +106,45 @@ export const ToolLine: FunctionComponent<ToolLineProps> = ({
...
+        style={{ minWidth: 40 }}
+        onClick={(event) => (event.shiftKey
+          ? onToggleShiftEntity(index, data, event)
+          : onToggleEntity(data, event))
+            }
+      >
+        <Checkbox
+          edge="start"
+          checked={
+                    (selectAll && !(data.id in (deSelectedElements || {})))
```
Comment (SarahBocognano) [inline opencti-platform/opencti-front/src/private/components/arsenal/tools/ToolLine.tsx]: indent

**Comment by Gwendoline-FAVRE-FELIX** [comment]
Normally, the indentations are checked.

**Comment by lndrtrbn** [comment]
> Normally, the indentations are checked.

To have better indentation in JSX when using condition like the ones point out by Sarah, use parenthesis around the right side of the condition. For example :
```
{!FAB_REPLACED
  && <Security needs={[KNOWLEDGE_KNUPDATE]}>
    <ChannelCreation paginationOptions={queryPaginationOptions} />
  </Security>
}
```
Will become :
```
{!FAB_REPLACED && (
  <Security needs={[KNOWLEDGE_KNUPDATE]}>
    <ChannelCreation paginationOptions={queryPaginationOptions} />
  </Security>
)}
```
And be linted automatically

**Comment by CelineSebe** [comment]
Massive operations are integrated into the new data table component, I think. I don't know if it was considered to use them here.

**Comment by Gwendoline-FAVRE-FELIX** [comment]
If it's functional, we'll merge it like that and look at the datatables later.

**Review by Archidoit — APPROVED** [review]

**Pull request merged**